### DISCUSSION TITLE
How run haddoc for an IHP-based project?

### DISCUSSION THREAD

**Discussion by michaelfliegner** [digitallyinduced/ihp#1129] [Q&A] [answered]
How run haddoc for an IHP-based project? Here's output for i.e.  Generated.Types:

`
mf@Michis:~/btgp-template$ haddock build/Generated/Types.hs 
Haddock coverage:

build/Generated/Types.hs:53:123: error:
    Pattern syntax in expression context: QueryBuilder.query@Userrole
    Did you mean to enable TypeApplications?
   |
53 |         let theRecord = User id email passwordHash lockedAt failedLoginAttempts (QueryBuilder.filterWhere (#refUser, id) (QueryBuilder.query @Userrole)) (QueryBuilder.filterWhere (#refUser, id) (QueryBuilder.query @Workflow)) def { originalDatabaseRecord = Just (Data.Dynamic.toDyn theRecord) }
   |                                                                                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

build/Generated/Types.hs:53:196: error:
    Pattern syntax in expression context: QueryBuilder.query@Workflow
    Did you mean to enable TypeApplications?
   |
53 |         let theRecord = User id email passwordHash lockedAt failedLoginAttempts (QueryBuilder.filterWhere (#refUser, id) (QueryBuilder.query @Userrole)) (QueryBuilder.filterWhere (#refUser, id) (QueryBuilder.query @Workflow)) def { originalDatabaseRecord = Just (Data.Dynamic.toDyn theRecord) }
   |                                                                                                                                                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

build/Generated/Types.hs:77:16: error:
    Operator applied to too few arguments: ?
   |
77 |     create :: (?modelContext :: ModelContext) => User -> IO User
   |                ^
mf@Michis:~/btgp-template$
 `

**Comment by mpscholten** [comment, marked as answer]
The error seems to be caused by missing to specify the required language extensions.

Try this:

```
haddock $(make print-ghc-extensions|tr ' ' '\n'|awk '{print "--optghc="$0}') --html build/Generated/Types.hs
```
